src/consumption.cpp

```diff
@@ -732,6 +732,19 @@ void player::consume_effects( item &food, bool rotten )
             fun = 1;
         }
     }
+    //Food is made out of meat (use ALLERGEN_MEAT, not CARNIVORE_OK (the latter has eggs and milk),
+    //Reduces bad morale penalties on food only made of meat.
+    if( has_trait( "CARNIVORE" ) && food.has_flag( "ALLERGEN_MEAT" ) &&
+        !food.has_any_flag( carnivore_blacklist ) &&
+        !( food.has_flag( "CANNIBALISM" ) && !has_trait( "CANNIBAL" ) ) ) {
+        if( fun < -2 ) {
+            fun = fun * 0.5;
+        } else if( fun > -2 && fun <= 3 ) {
+            fun = fun + 4;
+        } else if( fun > 3 ) {
+            fun = fun * 2;
+        }
```

Comment:
You can skip the first check and instead write it in one line like:
```
fun = std::max( fun + 4, fun * 2)
```

While this would make really horrible food less appealing, this is a tiny detail and sometimes clear code is better than extra features.